README

                             IO-Compress-Base

                             Version 2.014

                            2nd September 2008

       Copyright (c) 2005-2008 Paul Marquess. All rights reserved.
          This program is free software; you can redistribute it
           and/or modify it under the same terms as Perl itself.

DESCRIPTION
-----------

This module is the base class for all IO::Compress and IO::Uncompress
modules.

PREREQUISITES
-------------

Before you can build IO-Compress-Base you need to have the following
installed on your system:

    * Perl 5.004 or better. 

BUILDING THE MODULE
-------------------

Assuming you have met all the prerequisites, the module can now be built
using this sequence of commands:

    perl Makefile.PL
    make
    make test

INSTALLATION
------------

To install IO-Compress-Base, run the command below:

    make install

TROUBLESHOOTING
---------------

FEEDBACK
--------

How to report a problem with IO-Compress-Base.

To help me help you, I need all of the following information:

 1. The Versions of everything relevant.
    This includes:

     a. The *complete* output from running this
     
            perl -V
     
        Do not edit the output in any way.
        Note, I want you to run "perl -V" and NOT "perl -v".
     
        If your perl does not understand the "-V" option it is too
        old. This module needs Perl version 5.004 or better.     

     b. The version of IO-Compress-Base you have. 
        If you have successfully installed IO-Compress-Base, this one-liner
        will tell you:

           perl -MIO::Compress::Base -e 'print qq[ver $IO::Compress::Base::VERSION\n]'

        If you are  running windows use this

           perl -MIO::Compress::Base -e "print qq[ver $IO::Compress::Base::VERSION\n]"

        If you haven't installed IO-Compress-Base then search IO::Compress::Base.pm
        for a line like this:

          $VERSION = "2.014" ;

 2. If you are having problems building IO-Compress-Base, send me a
    complete log of what happened. Start by unpacking the IO-Compress-Base
    module into a fresh directory and keep a log of all the steps

        [edit config.in, if necessary]
        perl Makefile.PL
        make
        make test TEST_VERBOSE=1          

Paul Marquess <pmqs@cpan.org>


Changes
CHANGES
-------

  2.014 2 September 2008

      * Makefile.PL
        Updated to check for indirect dependencies.

  2.013 18 July 2008

      * IO::Compress::Base 
        - Allow IO::Compress::Base::Parameters::parse to accept an IO::Compress::Base::Parameters object.

  2.012 15 July 2008

      * IO::Compress::Base 
        - Silenced an uninitialised value warning when reading a line
          at a time from a zip file where the content uses ZIP_CM_STORE. 
          [Problem spotted & fixed by Jeff Holt]

      * IO::Compress::Base & IO::Uncompress::Base
        - local-ise $!, $? et al in the DESTROY methods.
          
  2.011 17 May 2008

      * IO::Compress::Base
        - Fixed problem that prevented the creation of a zip file that
          contained more than one compression method.

      * IO::Compress::Base::Common
        - The private Validator class in this module clashes with another
          CPAN module. Moved Validator into the IO::Compress::Base::Common
          namespace.
          [RT #35954]

  2.010 5 May 2008

      * Fixed problem that meant Perl 5.10 could not upgrade this module.
        [RT #35342]

  2.009 20 April 2008

      * Removed the alpha status from File::GlobMapper
      
      * IO::Compress::Base 
        When writing output never output a zero length buffer.
        Done to improve interoperability with other tied filenandle
        modules.

      * Changed IO::Uncompress::Base to not use the offset parameter of
        the read method when reading from a filehandle. 

        The object returned from Net::FTP::retr implements a non-standard
        read method. The third parameter is used for a timeout value
        rather than an offset.  
        [rt.cpan#33231]
        
      * Changed IO::Uncompress::Base to not use the eof method when
        reading from a filehandle. 

        The object returned from Net::FTP::retr implements both the read
        method and the eof method. Unfortunately the implementation of
        the read method uses non-buffered IO (by using sysread) while
        the eof method uses buffered IO. Mixing buffered and non-buffered
        IO results in data corruption.

  2.008 2 November 2007

      * Minor documentation changes in README

  2.006 1 September 20007

      * Makefile.PL
        Added INSTALLDIRS directive to install as a core module when built
        on a perl >= 5.9.

  2.005 18 June 2007

      * Stephen Turner reported a problem when using IO::Uncompress::Gunzip
        with XML::Parser. Turns out there were two issues.  

        Firstly an IO::Uncompress object isn't an IO::Handle. It is now.

        Secondly the implementation of "read" wasn't honouring this 

            SCALAR will be grown or shrunk to the length actually read.

        In particular it didn't do the right thing on EOF.
        This has been fixed.

  2.004 3 March 2007

      * Made seek less wasteful of memory.

  2.003 2 January 2007

      * Added explicit version checking

  2.002 29 December 2006

      * Documentation updates.

      * Added IO::Handle to the ISA test in isaFilehandle

      * Add an explicit use_ok test for Scalar::Util in the test harness.
        The error message reported by 01misc implied the problem was
        somewhere else.
        Also explictly check that 'dualvar' is available.

  2.001 1 November 2006

      * Remove beta status.

  2.000_14 26 October 2006

      * IO::Uncompress::Base
        Added support for $/ in record mode

      * IO::Uncompress::Base
        The readline interface was substantially slower than the 1.x
        equivalent. This has now been sorted. 
        Thanks to Andreas J. Koenig for spotting the problem.

      * IO::Uncompress::AnyUncompress
        Added IO::Uncompress::Lzf to the list of supported uncompresors.

      * IO::Uncompress::Base
        Added TrailingData to one-shot interface.

      * IO::Uncompress::AnyUncompress
        Remove raw-deflate (RFC1951) from the default list of compressors 
        to check.
        It can still be included if the new RawInflate parameter is
        supplied.
        This change was made because the only way to tell if content is 
        raw-deflate is to attempt to uncompress it - a few false positives
        have popped up recently, which suggests that auto-detecting raw 
        deflate is far from perfect.
        The equivalent change has been made to IO::Uncompress::AnyInflate.
        [Core patch #28445]

      * Don't check that filehandles are writable. It would seem that 
        "-w *STDOUT" on windows returns false. 
        [Core Patch #28415]

  2.000_13 20 June 2006

      * Store compress & uncompressed sizes as 64-bit.

      * For one-shot uncompression, like this

            unzip "some.zip" => \@a, MultiStream => 1;

        Push each uncompressed stream from "some.zip" onto @a.

      * Added IO::Compress::Base::FilterEnvelope

      * Added IO::Uncompress::Base::nextStream

      * The '-' filehandle now maps to either *STDIN or *STDOUT. 
        This keeps mod_perl happier. Was using these before

            new IO::File("<-")
            new IO::File(">-")
  
  2.000_12 3 May 2006

  2.000_11 10 April 2006

      * Transparent + InputLength made more robust where input data is not
        compressed.

  2.000_10 13 March 2006

      * AnyUncompress doesn't assume  that IO-Compress-Zlib is installed any
        more.

  2.000_09 3 March 2006

      * Released to CPAN.

  2.000_08 2 March 2006

      * Split IO::Compress::Base into its own distribution.

      * Added opened, autoflush and input_line_number.

      * Beefed up support for $.



Makefile.PL
#! perl -w

use strict ;
require 5.004 ;

$::VERSION = '2.014' ;

use private::MakeUtil;
use ExtUtils::MakeMaker 5.16 ;

UpDowngrade(getPerlFiles('MANIFEST')) 
    unless $ENV{PERL_CORE};

# Modules supplied by this distribution    
my @myModules = qw( IO::Compress::Base
                    IO::Uncompress::Base
                    IO::Compress::Base::Common
                  );

# Direct dependencies                  
my @MainPREREQ = qw();

# Get indirect dependencies that will break if we install this version
push @MainPREREQ, FindBrokenDependencies($::VERSION, @myModules)
    unless $ENV{PERL_CORE} ;

my %MainPREREQ = map { $_ => $::VERSION } @MainPREREQ; 
    
WriteMakefile( 
    NAME         => 'IO::Compress::Base',
    VERSION_FROM => 'lib/IO/Compress/Base.pm',
    'dist'       => { COMPRESS     => 'gzip', 
                      TARFLAGS     => '-chvf',
                      SUFFIX       => 'gz',
                      DIST_DEFAULT => 'MyTrebleCheck tardist',
                    },

    (
      $ENV{SKIP_FOR_CORE}
        ? (MAN3PODS    => {})
	    : (PREREQ_PM   => { %MainPREREQ,
                            'Scalar::Util'  => 0,
		                    $] >= 5.005 && $] < 5.006   
                                ? ('File::BSDGlob' => 0) 
                                : () }
	      )
    ),
       
    (
    $] >= 5.005
        ? (ABSTRACT_FROM => 'lib/IO/Compress/Base.pm',
            AUTHOR       => 'Paul Marquess <pmqs@cpan.org>')
        : ()
    ),

    INSTALLDIRS => ($] >= 5.009 ? 'perl' : 'site'),

    (
    $] >= 5.009 && ! $ENV{PERL_CORE}
        ? (INST_LIB    => 'blib/arch')
        : ()
    ),

    ((ExtUtils::MakeMaker->VERSION() gt '6.30') ?
        ('LICENSE'  => 'perl')         : ()),
    
) ;

# end of file Makefile.PL


